core/metacling/src/TClingTypeInfo.h

```diff
@@ -42,6 +44,8 @@ class TClingTypeInfo {
 private:
    cling::Interpreter  *fInterp;    //Cling interpreter, we do *not* own.
    clang::QualType      fQualType;  //Clang qualified type we are querying.
+   // FIXME: Factor out in the TClingGenericInfo or a similar concept.
+   mutable std::string fNameCache;
```

Comment:
Given that Name() is no longer const, do you still need this to be mutable?

Reply:
This is not. The TClingGenericInfo seems to be useful when working with decls. This works on types. We will need to have something similar there.

Reply:
still it had seems that the Name() for that class was non-const.  But it is also seems you removed this part, so never mind:)